README.org

```diff
@@ -0,0 +1,47 @@
+[[https://travis-ci.org/threatgrid/ring-turnstile-middleware][https://travis-ci.org/threatgrid/ring-turnstile-middleware.png?branch=master]]
+
+* ring-turnstile-middleware
+
+Ring middleware for the Turnstile rate limiting service
+
+** Usage
+
+Use =wrap-rate-limit-fn= to create an instance of the middleware, wrap your routes with it:
+
+#+BEGIN_SRC clojure
+
+(require '[ring.middleware.turnstile :refer [wrap-rate-limit-fn]]
+         '[ring.util.response :refer [response]])
+
+(def handler [request]
+  (response {:foo "bar"}))
+
+(def app
+  (let [wrap-rate-limit
+        (wrap-rate-limit-fn {:exceptions [{:ip "127.0.0.1"}]
+                             :limits [{:key [:ip :method]
+                                       :default 240
+                                       :override {["10.2.34.5" "GET"] 1000}}]}
```

Comment:
I would suggest you provide a function instead of a configuration map as the first param of the middleware. And you should provide a default fn to be used. That way you don't force any configuration language/logic to the user.